app/controllers/templates_controller.rb

```diff
@@ -78,8 +78,26 @@ def auto_complete_controller_name
     type_name_plural
   end
 
+  def preview
+    base = Host.authorized(:view_hosts)
+    @host = params[:preview_host_id].present? ? base.find(params[:preview_host_id]) : base.first
+    @template.template = params[:template]
+    safe_render(@template)
+  end
+
   private
 
+  def safe_render(template)
+    begin
+      load_template_vars
+      render :text => unattended_render(template) and return
+    rescue => error
+      msg = _("There was an error rendering the %s template: ") % (template.name)
```

Comment:
I think I saw we have this format elsewhere, but I'd suggest avoiding translating messages that are logged, and to include the %s for the error.message in the translated string - don't just concatenate.